assets/components/textInput/textInput.jsx

```diff
@@ -36,30 +37,45 @@ type PropTypes = {
   pattern?: string,
   modifierClasses: Array<?string>,
   onBlur: () => void,
+  pattern?: string,
+  type: ?string,
+  showError: boolean,
+  errorMessage: string,
 };
 
 
 // ----- Component ----- //
 
 export default function TextInput(props: PropTypes) {
 
+  const { modifierClasses } = props;
+  if (props.showError) {
+    modifierClasses.push('error');
+  }
+
   return (
-    <div className={classNameWithModifiers('component-text-input', props.modifierClasses)}>
-      <label htmlFor={props.id} className="component-text-input__label">
-        {props.labelText}
-      </label>
-      <input
-        className={classNameWithModifiers('component-text-input__input', props.modifierClasses)}
-        type={props.type}
-        id={props.id}
-        onChange={event => props.onChange(event.target.value || '')}
-        onBlur={props.onBlur}
-        value={props.value}
-        placeholder={props.placeholder}
-        required={props.required}
-        autoComplete={props.autocomplete}
-        autoCapitalize={props.autocapitalize}
-        pattern={props.pattern}
+    <div className="component-text-input__container">
+      <div className={classNameWithModifiers('component-text-input', modifierClasses)}>
+        <label htmlFor={props.id} className="component-text-input__label">
+          {props.labelText}
+        </label>
+        <input
+          className={classNameWithModifiers('component-text-input__input', modifierClasses)}
+          type={props.type}
+          pattern={props.pattern}
+          id={props.id}
+          onChange={event => props.onChange(event.target.value || '')}
+          onBlur={props.onBlur}
+          value={props.value}
+          placeholder={props.placeholder}
+          required={props.required}
```

Comment:
maybe we could remove these since they currently do nothing? could be misleading for a future developer (they might think this would cause the field to validate when it won't)

alternatively we could try and have these automatically respected by our validation logic but I suspect that might be a pain

Reply:
I've removed them for now.